python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -148,6 +148,22 @@ def get_lsst_photometric_uncertainties(self):
                                                  'lsstBandpassDict')
 
 
+    def calculateVisibility(self, magFilter, m5, sigma=0.12):
+        """
+        Calculate the probability of detecting a particular source.
+
+        @ param [in] magFilter is the magnitude of the object in the given filter.
+
+        @ param [in] obs_metadata is the observation metadata for a given visit (to determine m5).
+
+        @ param [out] visibility (0/1).
+        """
+        if len(magFilter) == 0:
+            return numpy.array([])
+        completeness = 1.0 / (1 + numpy.exp((magFilter - m5)/sigma))
+        probability = numpy.random.random_sample(len(magFilter))
```

Comment:
That's a good point. We could make the SQL return everything in order if we sorted by objId? But would that make things too slow? 
I would like to make it optionally at least potentially repeatable -- what do you do for setting random seeds in other instance catalogs?